src/blst-native/publicKey.ts

```diff
@@ -18,15 +18,19 @@ export class PublicKey implements IPublicKey {
     return this.fromBytes(hexToBytes(hex));
   }
 
-  static aggregate(publicKeys: PublicKey[]): PublicKey {
+  static aggregate(publicKeys: PublicKeyArg[]): PublicKey {
     if (publicKeys.length === 0) {
       throw new EmptyAggregateError();
     }
 
-    const pk = blst.aggregatePublicKeys(publicKeys.map(({value}) => value));
+    const pk = blst.aggregatePublicKeys(publicKeys.map(PublicKey.convertToBlstPublicKeyArg));
     return new PublicKey(pk);
   }
 
+  static convertToBlstPublicKeyArg(publicKey: PublicKeyArg): blst.PublicKeyArg {
+    return publicKey instanceof PublicKey ? publicKey.value : (publicKey as Uint8Array);
```

Comment:
we don't have to cast here given the definition of `PublicKeyArg`
```typescript
export type PublicKeyArg = PublicKey | Uint8Array;
```

Reply:
Its because the TSC is getting confused between the IPublicKey and PublicKey.  PublicKeyArg is `IPublicKey` not the hard class that implements it.

<img width="722" alt="Screenshot 2024-04-22 at 1 47 22 PM" src="https://github.com/ChainSafe/bls/assets/18608739/11a5568e-dd1e-486c-9054-4fe5a531c4f1">

<img width="736" alt="Screenshot 2024-04-22 at 1 41 33 PM" src="https://github.com/ChainSafe/bls/assets/18608739/ffde8218-2147-4044-8a21-a79836b29eb6">

<img width="691" alt="Screenshot 2024-04-22 at 1 42 12 PM" src="https://github.com/ChainSafe/bls/assets/18608739/aeb8ff02-c2dd-49fb-bd67-7bb971b2f2f0">

There are a couple of way I cam up to solve this.
1) Create a separate PublicKeyArg that lives in the implementation file.

<img width="687" alt="Screenshot 2024-04-22 at 1 47 00 PM" src="https://github.com/ChainSafe/bls/assets/18608739/578a5a38-aff8-40a2-984c-a774ca6f8785">

<img width="750" alt="Screenshot 2024-04-22 at 1 46 31 PM" src="https://github.com/ChainSafe/bls/assets/18608739/b2d4af25-4d1c-455e-abe8-29ee3cc3819d">

<img width="741" alt="Screenshot 2024-04-22 at 1 46 49 PM" src="https://github.com/ChainSafe/bls/assets/18608739/b88f27a5-90b8-452e-8959-09abb9a1723b">

2) Using the static casts in a few places and making a comment about why the static cast was present.  As a note that location you noticed was one I didn't add the not but can add it in like below.

<img width="727" alt="Screenshot 2024-04-22 at 1 51 27 PM" src="https://github.com/ChainSafe/bls/assets/18608739/ac968d63-5a0b-49f3-bd0a-63aa4f81413f">

<img width="661" alt="Screenshot 2024-04-22 at 1 52 36 PM" src="https://github.com/ChainSafe/bls/assets/18608739/cc3331ee-02e2-41d0-94e2-1d50b514e453">


In the end I went with option 2 because I thought it would be more confusing to have multiple types that looked the same but were not floating around.  If the wrong one was "auto imported" it would be confusing to debug for someone without history with the repo.

I switched that conditional around so that its PublicKey that gets cast and I added the comment to that location.  Let me know if you want me to go another direction though and Ill be happy to switch it up.

```ts
  static convertToBlstPublicKeyArg(publicKey: PublicKeyArg): blst.PublicKeyArg {
    // need to cast to blst-native key instead of IPublicKey
    return publicKey instanceof Uint8Array ? publicKey : (publicKey as PublicKey).value;
  }
```

Reply:
ah I see, switching makes it easier to reason about 👍